lib/workers/pr/body/updates-table.ts

```diff
@@ -56,7 +58,7 @@ export function getPrUpdatesTable(config: BranchConfig): string {
     }
     return res;
   });
-  const tableColumns = getNonEmptyColumns(tableDefinitions, tableValues);
+  const tableColumns = getNonEmptyColumns(config.prBodyColumns, tableValues);
```

Comment:
Hi @rarkins @viceice, 
i supposed that the config.prBodyColumns has all the columns that are supposed to appear in the PR, regardless if they are empty/full, is that assumption correct?

Reply:
Yes. Later any empty columns are then pruned

Reply:
I'm asking because i can see that there are 2 types of prBodyColumns, one inside the config, and one inside each Upgrade, 
so the actual question is, could it be that one upgrade has different prBodyColumns than the config.prBodyColumns?

Reply:
Well that was the main question, is config.prBodyColumns an union of all columns that should appear? 

if i want to check all upgrade.prBodyDefinition and union all fields, doesn't that cause a problem in sorting the column headers? there could be different rows for each upgrade no? what do i take first and last? 

is there a way to set the prBodyColumns to have an union before we reach that point? without the sorting problem?

what if we did more than 1 table in the PR is that possible in the current design? i mean 1 table for each updateType then we wouldn't have to worry about this. xD

Reply:
i cant Union anything since the prBodyColumns doesnt exist for each upgrade..

so basically i shouldnt use "config.prBodyColumns", since its for the first upgrade? that makes it useless..

i should just take ALL the Object.Keys(config.prbodyDefinitions) since they appear to always be "all headers available" and use them instead?

Reply:
@PhilipAbed which problem are you trying to solve?

1. The list of columns is wrong? or
2. The content of some rows is wrong?

Because when you talk about unioning prBodyColumns, it sounds like question (1), while I thought this issue was about (2)

Reply:
Back to the prBodyColumns, you can union and deduplicate them with this code: `config.prBodyColumns = [...new Set(config.upgrades.reduce((existing, upgrade) => existing.concat(upgrade.prBodyColumns), []))];`.

This logic could be added somewhere around here: https://github.com/renovatebot/renovate/blob/427df6b7bc875f7270a6a26bfc14a97708f06802/lib/workers/repository/updates/generate.ts#L298-L323

Reply:
Thanks! i changed it.